addons/web/static/src/search/control_panel/control_panel.xml

```diff
@@ -61,33 +61,26 @@
     <t t-name="web.ControlPanel.Small" owl="1">
         <t t-if="state.showMobileSearch">
             <t t-portal="'body'">
-                <div class="o_searchview o_mobile_search">
-                    <div class="o_mobile_search_header">
-                        <button
-                            type="button"
-                            class="o_mobile_search_button btn"
-                            t-on-click="() => state.showMobileSearch = false"
-                        >
-                            <i class="fa fa-arrow-left"/>
-                            <strong class="ms-2">FILTER</strong>
-                        </button>
-                        <button
-                            type="button"
-                            class="o_mobile_search_button btn"
-                            t-on-click="() => env.searchModel.clearQuery()"
-                        >
-                            CLEAR
-                        </button>
-                    </div>
+                <div class="o_mobile_search  position-fixed bottom-0 start-0 end-0 w-100 d-flex flex-column p-0 overflow-auto bg-view border-0">
                     <div class="o_mobile_search_content">
                         <t t-if="display['top-right']">
-                            <t t-slot="control-panel-top-right">
+                            <t t-slot="control-panel-top-right d-flex">
+                                <button type="button" class="o_enable_searchview o_mobile_search_back btn btn-link fa fa-arrow-left position-absolute start-0 z-index-1" t-on-click="() => state.showMobileSearch = false"/>
                                 <SearchBar/>
                             </t>
                         </t>
+                        <div class="bg-100 border-bottom text-end" t-if="env.searchModel.facets.length !== 0">
```

Comment:
Convention: place `t-if` attributes first. The fact that the element is rendered conditionally is more important that the style it will have  ;)